toolkit/tools/pkg/imagecustomizerlib/customizeverity.go

```diff
@@ -0,0 +1,222 @@
+	err = updateMarinerCfgWithInitramfs(imageChroot)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func buildDracutModule(dracutModuleName string, imageChroot *safechroot.Chroot) error {
+	var err error
+
+	// This function will be run inside the chroot to list kernel files in /boot
+	listKernels := func() ([]string, error) {
+		var kernels []string
+		// Assuming the vmlinuz files are located in /boot
+		files, err := filepath.Glob("/boot/vmlinuz-*")
+		if err != nil {
+			return nil, err
+		}
+		for _, file := range files {
+			kernels = append(kernels, filepath.Base(file))
+		}
+		return kernels, nil
+	}
+
+	var kernelFiles []string
+	err = imageChroot.Run(func() error {
+		kernelFiles, err = listKernels()
+		return err
+	})
+	if err != nil {
+		return fmt.Errorf("failed to list kernels in chroot: %w", err)
+	}
+
+	if len(kernelFiles) == 0 {
+		return fmt.Errorf("no kernels found in chroot environment")
+	}
+
+	// Extract the version from the kernel filename (e.g., vmlinuz-5.15.131.1-2.cm2 -> 5.15.131.1-2.cm2)
+	kernelVersion := strings.TrimPrefix(kernelFiles[0], "vmlinuz-")
+
+	buildDracutModuleArgs := []string{
+		"dracut", "-f", "--kver", kernelVersion, "-a",
+		// Placeholder for dracut module name.
+		"",
+	}
+
+	buildDracutModuleArgs[len(buildDracutModuleArgs)-1] = dracutModuleName
+
+	err = imageChroot.Run(func() error {
+		_, _, err = shell.Execute("sudo", buildDracutModuleArgs...)
+		return err
+	})
+	if err != nil {
+		return fmt.Errorf("failed to build dracut module - (%s):\n%w", dracutModuleName, err)
+	}
+
+	return nil
+}
+
+func updateMarinerCfgWithInitramfs(imageChroot *safechroot.Chroot) error {
+	initramfsPath := filepath.Join("boot", "initramfs-*")
+
+	// Fetch the initramfs file name.
+	var initramfsFiles []string
+	err := imageChroot.Run(func() error {
+		var innerErr error
+		initramfsFiles, innerErr = filepath.Glob(initramfsPath)
+		return innerErr
+	})
+	if err != nil {
+		return fmt.Errorf("failed to list initramfs file: %w", err)
+	}
+
+	// Ensure an initramfs file is found
+	if len(initramfsFiles) != 1 {
+		return fmt.Errorf("expected one initramfs file, but found %d", len(initramfsFiles))
+	}
+
+	newInitramfs := filepath.Base(initramfsFiles[0])
+
+	cfgPath := filepath.Join("boot", "mariner.cfg")
+
+	// Update mariner.cfg to reference the new initramfs
+	err = imageChroot.Run(func() error {
+		input, innerErr := ioutil.ReadFile(cfgPath)
+		if innerErr != nil {
+			return fmt.Errorf("failed to read mariner.cfg: %w", innerErr)
+		}
+
+		lines := strings.Split(string(input), "\n")
+		for i, line := range lines {
+			if strings.HasPrefix(line, "mariner_initrd=") {
+				lines[i] = "mariner_initrd=" + newInitramfs
+			}
+		}
+		output := strings.Join(lines, "\n")
+		return ioutil.WriteFile(cfgPath, []byte(output), 0644)
+	})
+
+	return nil
+}
+
+func updateGrubConfig(resolvedVerityDevice string, resolvedHashDevice string, salt string, rootHash string, verityCorruptionResponse string) error {
```

Comment:
When updating the linux command line in grub.cfg, I suggest to follow the pattern seen in the 2.0 grub.cfg for systemd:

- https://github.com/microsoft/cbl-mariner/blob/main/toolkit/tools/internal/resources/assets/grub2/grub.cfg#L6-L8
- https://github.com/microsoft/cbl-mariner/blob/main/SPECS/systemd/systemd.cfg#L2

Here's my suggestion. Add this if-else block to the base grub.cfg.

```
if [ -f  $bootprefix/verity.cfg ]; then
	load_env -f $bootprefix/verity.cfg
else
	set verity_cmdline=
fi
```

Add $verity_cmdline to the end of the linux command in the base grub.cfg too.

Then, define a template and have this function (over)write it to $bootprefix/verity.cfg:

```
rd.systemd.verity=1 roothash={{.RootHash}} systemd.verity_root_data={{.SystemdVerityRootData}} systemd.verity_root_hash={{.SystemdVerityRootHash}} systemd.verity_root_options={{.SystemdVerityRootOptions}},salt={{.Salt}}
```

This makes would make this function simpler and idempotent, but requires a small change to the grub.cfg of the base image.

Reply:
We need to add directly into the Kernel cmdline argument since Trident needs to consume in the later workflow.